lib/bookmark_manager.rb

```diff
@@ -157,4 +181,31 @@ def update_topic_user_bookmarked(topic, opts = {})
   def update_user_option(bookmark)
     @user.user_option.update!(bookmark_auto_delete_preference: bookmark.auto_delete_preference)
   end
+
+  def after_create_bookmarkable_post(post, opts = {})
+    update_topic_user_bookmarked(post.topic, opts)
+  end
+
+  def after_create_bookmarkable_topic(topic, opts = {})
+    update_topic_user_bookmarked(topic, opts)
+  end
+
+  def after_destroy_bookmarkable_post(bookmark)
+    { topic_bookmarked: update_topic_user_bookmarked(bookmark.bookmarkable.topic) }
```

Comment:
I feel like it isn't this method's responsibility to know what the return object should be for another method.

Reply:
Hmm I see what you mean, I guess we can just remove this return value in the manager completely and just query it elsewhere from whatever is destroying the bookmark (the controller)